src/AppInstallerCLICore/Workflows/PinFlow.cpp

```diff
@@ -0,0 +1,247 @@
+    void RemovePin(Execution::Context& context)
+    {
+        auto package = context.Get<Execution::Data::Package>();
+        std::vector<Pinning::Pin> pins;
+
+        // TODO: We should support querying the multiple sources for a package, instead of just one
+        auto availableVersion = package->GetLatestAvailableVersion();
+
+        auto pinningIndex = context.Get<Execution::Data::PinningIndex>();
+        Pinning::PinKey pinKey{
+            availableVersion->GetProperty(PackageVersionProperty::Id).get(),
+            availableVersion->GetProperty(PackageVersionProperty::SourceIdentifier).get() };
+        AICLI_LOG(CLI, Info, << "Removing pin for package [" << pinKey.PackageId << "] from source [" << pinKey.SourceId << "]");
+        if (!pinningIndex->GetPin(pinKey))
+        {
+            AICLI_LOG(CLI, Warning, << "Pin does not exist");
+            context.Reporter.Warn() << Resource::String::PinDoesNotExist(pinKey.PackageId) << std::endl;
+            AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_PIN_DOES_NOT_EXIST);
+        }
+
+        pinningIndex->RemovePin(pinKey);
+    }
+
+    // Report the pins in a table.
+    // Required Args: None
+    // Inputs: Pins
+    // Outputs: None
+    void ReportPins(Execution::Context& context)
+    {
+        const auto& pins = context.Get<Execution::Data::Pins>();
+        if (pins.empty())
+        {
+            context.Reporter.Info() << Resource::String::PinNoPinsExist << std::endl;
+            return;
+        }
+
+        // TODO: Use package and source names
+        Execution::TableOutput<4> table(context.Reporter,
+            {
+                Resource::String::SearchId,
+                Resource::String::SearchSource,
+                Resource::String::SearchVersion,
+                Resource::String::PinType,
+            });
+
+        for (const auto& pin : pins)
+        {
+            // TODO: Avoid these conversions to string
+            table.OutputLine({
+                pin.GetPackageId(),
+                std::string{ pin.GetSourceId() },
+                std::string{ pin.GetVersionString() },
+                std::string{ ToString(pin.GetType()) },
+                });
+        }
+
+        table.Complete();
+    }
+
+    // Resets all the existing pins.
+    // Required Args: None
+    // Inputs: PinningIndex
+    // Outputs: None
+    void ResetAllPins(Execution::Context& context)
+    {
+        AICLI_LOG(CLI, Info, << "Resetting all pins");
+        if (context.Args.Contains(Execution::Args::Type::Force))
+        {
+            context.Reporter.Info() << Resource::String::PinResettingAll << std::endl;
+
+            if (context.Get<Execution::Data::Pins>().empty())
+            {
+                context.Reporter.Info() << Resource::String::PinNoPinsExist << std::endl;
+            }
+            else
+            {
+                auto pinningIndex = context.Get<Execution::Data::PinningIndex>();
+                pinningIndex->ResetAllPins();

```

Comment:
From looking at PinCommand implementation, seems like we intend to support resetting filtered by source, but this ResetAllPins() just resets everything.

Reply:
Good catch :)
Added filtering by source